lib/flash.js

```diff
@@ -0,0 +1,184 @@
+var async = require('async');
+var PAGE = 256;
+var logs = require('./logs');
+var CHIP_ID = new Buffer([0x01, 0x02, 0x19]);
+
+function address(addr) {
+  return [(addr >> 24) & 0xFF, (addr >> 16) & 0xFF, (addr >> 8) & 0xFF, addr & 0xFF];
+}
+
+// Generate a mediatek factory partition
+function factoryPartition(mac1, mac2) {
+  return concatToBuffer(
+    [0x20, 0x76, 0x03, 0x01],
+    mac1,
+    Array(30).fill(0xff),
+    mac2
+  );
+}
+
+function transaction(usbConnection, write, read, status_poll, wren, next) {
+  read = read || 0;
+  status_poll = status_poll || false;
+  wren = wren || false;
+
+  if (write.length > 500 || read >= Math.pow(2, 24)) {
+    return next(new Error('Transaction too large'));
+  }
+  var flags = (status_poll ? 1 : 0) | ((wren ? 1 : 0) << 1);
+  var hdr = [(read >> 0) & 0xff, (read >> 8) & 0xff, (read >> 16) & 0xff, flags];
+
+  var data = concatToBuffer(hdr, write);
+
+  usbConnection.epOut.transfer(data, function(err) {
+    if (err) {
+      return next(err);
+    }
+    if (read > 0) {
+      return usbConnection.epIn.transfer(read, next);
+    }
+    next();
+  });
+}
+
+function readChipId(usbConnection, next) {
+  transaction(usbConnection, [0x9f], 3, null, null, next);
+}
+
+function checkChipId(chipId, next) {
+  if (Buffer.compare(chipId, CHIP_ID) !== 0) {
+    return next(new Error('Invalid chip ID (flash communication error)'));
```

Comment:
- In what case would this error get thrown? Why would the `chipId` not match?
- What does this mean to someone using this command on their Tessel? 
- Is there a way to resolve it?